### ISSUE TITLE
[onert] Let's introduce `onert_train` tool

### ISSUE THREAD

**Issue by jyoungyun** [Samsung/ONE#10804]
### What

Let's introduce `onert_train` tool.

### Why

To train a model, we need to receive additional parameters from the user.
It is the start point of training tool.

**Comment by glistening** [comment]
@jyoungyun Why don't you just add new parameters in existing `onert_run`? I think it introduce many code duplication as in #10782. We had similar experience last year when `npud` is introduced.

**Comment by jyoungyun** [comment]
In order to implement the training function, there are many specific options such as Learning Rate, Optimizer, Batchsize, Loss function, etc, which makes complicate the the existing `onert_run` code. In addition, the `onert_train` will always use `training` backend. However, `onert_run` supports selectiing a backend from the user. Considering many things, it would be better to add a new executable for training feature.

**Comment by glistening** [comment]
@jyoungyun I think all parameters you listed will be a part of circle+. I think `onert_run` can get the values as whole in a kind of config file.

**Comment by jyoungyun** [comment]
@glistening I worried about code duplication. So I will effort to reduce it. However, our basic direction for the training feature was not touch the existing code as much as possible. In order to do so, code duplication can inevitably occur.

If you don't agree to introduce `onert_train`, I will fix the `onert_run` code directly.

**Comment by jyoungyun** [comment]
We talked about this issue offline, and the directory under the `tests` directory does not include release criteria. Therefore, this code will add a separate tool as `onert_train`. (Apart from this decision, I will still work to reduce the code duplication.)
